package org.android.bbangzip.data.repositoryImpl.remote

import org.android.bbangzip.data.datasource.remote.DummyRemoteDataSource
import org.android.bbangzip.domain.model.DummyEntity
import org.android.bbangzip.domain.repository.remote.DummyRepository
import javax.inject.Inject

class DummyRepositoryImpl
@Inject
constructor(
    private val dummyRemoteDataSource: DummyRemoteDataSource,
) : DummyRepository {
    override suspend fun fetchDummy(dummyId: Long): Result<DummyEntity> =
        runCatching {
            dummyRemoteDataSource.getDummy(dummyId = dummyId).data.toDummyEntity()
        }
}
